kolibri/core/device/api.py

```diff
@@ -312,6 +321,35 @@ def get_queryset(self):
             return UserSyncStatus.objects.none()
         return UserSyncStatus.objects.all()
 
+    def has_downloads(self, obj):
+        is_having_downloads = Subquery(
```

Comment:
First, because of my comments regarding `serializers.SerializerMethodField()`, I do not think this code is being executed by your tests or calls to this API.

Also, this API is used for both coaches and learners. My suggestion was to use queryset annotations (see the `annotate_queryset` method) for extracting this information. Using individual methods for this means your approach would invoke them for each object in the result set, which isn't ideal for the coach use case since there could be many learners in the result set. The way I see it, you can go either way (individual methods vs annotations), but with either I think it makes sense to conditionalize this such the request for a single learner's status only returns these *download fields to limit the overhead since they won't be utilized on the coach side. It is likely easier to conditionalize annotations since they would inherently be removed from the resulting query if the query values do not include the field.
1) If you choose to use individual methods: There are a few different ways to achieve this overall, one might be to use the `field_map` to call the methods. Additionally, in this case these shouldn't be using `Subquery` as that is only valid within another queryset.
2) As an annotation, this should use `Exists` instead of `Subquery` since we want the result to be a boolean.